### ISSUE TITLE
the background operator doesn't always issue a parse error when given an expression with no side effects

### ISSUE THREAD

**Issue by davidnich** [qorelanguage/qore#2747]
ex:
```
david@greybeard:~/src/qore/git/qore/build$ qore -nX 'background sub() {}'
2
```

nothing is done except that the thread is started - the closure is not executed

**Comment by davidnich** [comment]
related to #2826

**Comment by martin-zemek** [comment]
Actually, no parse error is issued even without the `background` operator:
`qore -ne 'sub() {};'`

**Comment by martin-zemek** [comment]
This issue has been fixed along with #2826
